### PR TITLE
Change dropdown type filter from Texture to Texture2D in certain nodes

### LINKED ISSUE #84104: LineEdit.right_icon is a Texture2D, but shows dropdown as if it's Texture
### Godot version

Godot v4.2.beta3 - Pop!_OS 22.04 LTS - X11 - Vulkan (Forward+) - integrated Intel(R) Graphics (ADL GT2) () - 12th Gen Intel(R) Core(TM) i5-1235U (12 Threads)

### System information

Godot v4.2.beta3 - Pop!_OS 22.04 LTS - X11 - Vulkan (Forward+) - integrated Intel(R) Graphics (ADL GT2) () - 12th Gen Intel(R) Core(TM) i5-1235U (12 Threads)

### Issue description

image

image

### Steps to reproduce

Click right_icon on LineEdit

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by BlueCube3310**
Fixes #84104

The aforementioned issue happens also with Decal and Sprite3D.

### CODE REVIEW

**Review by KoBeWi — APPROVED** [review]

**Comment by AThousandShips** [comment]
Needs to resolve compatibility

File: misc/extension_api_validation/4.0-stable_4.1-stable.expected [hunk 1]
```diff
@@ -349,3 +349,14 @@ Validate extension JSON: Error: Hash changed for 'classes/EditorUndoRedoManager/
 Validate extension JSON: Error: Hash changed for 'classes/UndoRedo/methods/create_action', from 0AEC1BFC to E87757EB. This means that the function has changed and no compatibility function was provided.
 
 Added a optional parameters with default values. No adjustments should be necessary.
+
+GH-84113
```
Comment (AThousandShips) [inline misc/extension_api_validation/4.0-stable_4.1-stable.expected:353]:
```suggestion

GH-84113
```
Two lines

File: misc/extension_api_validation/4.1-stable.expected [hunk 2]
```diff
@@ -293,3 +293,14 @@ GH-79527
 Validate extension JSON: Error: Field 'classes/ParticleProcessMaterial/properties/orbit_velocity_curve': type changed value in new API, from "CurveTexture" to "CurveTexture,CurveXYZTexture".
 
 Added accepted curve type from only CurveTexture to CurveXYZTexture.
+
+GH-84113
```
Comment (AThousandShips) [inline misc/extension_api_validation/4.1-stable.expected:297]:
```suggestion

GH-84113
```

File: misc/extension_api_validation/4.0-stable_4.1-stable.expected
Comment (AThousandShips) [inline misc/extension_api_validation/4.0-stable_4.1-stable.expected:1]: Do not add to this file, it's for 4.0->4.1

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**